router/dataplane.go

```diff
@@ -1946,26 +1946,37 @@ func (p *scionPacketProcessor) packSCMP(
 func (p *scionPacketProcessor) parsePath( /*@ ghost ub []byte, ghost dp io.DataPlaneSpec @*/ ) (respr processResult, reserr error) {
 	var err error
 	// @ unfold acc(p.scionLayer.Mem(ub), R6)
-	// @ defer fold acc(p.scionLayer.Mem(ub), R6)
 	// @ ghost startP := p.scionLayer.PathStartIdx(ub)
 	// @ ghost endP := p.scionLayer.PathEndIdx(ub)
 	// @ ghost ubPath := ub[startP:endP]
 	// @ sl.SplitRange_Bytes(ub, startP, endP, R2)
-	// @ ghost defer sl.CombineRange_Bytes(ub, startP, endP, R2)
+	// @ assert acc(p.path.Mem(ubPath), R20)
 	p.hopField, err = p.path.GetCurrentHopField( /*@ ubPath @*/ )
+	// @ assert err == nil ==> p.path.CorrectlyDecodedHF_RawOffsetIndex(p.hopField, ubPath)
 	// @ fold p.d.validResult(processResult{}, false)
 	if err != nil {
 		// TODO(lukedirtwalker) parameter problem invalid path?
+		// @ fold acc(p.scionLayer.Mem(ub), R6)
+		// @ ghost sl.CombineRange_Bytes(ub, startP, endP, R2)
 		return processResult{}, err
 	}
+	// @ assert acc(p.path.Mem(ubPath), R20)
 	p.infoField, err = p.path.GetCurrentInfoField( /*@ ubPath @*/ )
 	if err != nil {
 		// TODO(lukedirtwalker) parameter problem invalid path?
+		// @ fold acc(p.scionLayer.Mem(ub), R6)
+		// @ ghost sl.CombineRange_Bytes(ub, startP, endP, R2)
 		return processResult{}, err
 	}
-	// @ TemporaryAssumeForIO(slayers.ValidPktMetaHdr(ub))
-	// @ TemporaryAssumeForIO(len(absPkt(dp, ub).CurrSeg.Future) > 0)
-	// @ TemporaryAssumeForIO(p.EqAbsHopField(absPkt(dp, ub)))
+	// @ assert acc(p.path.Mem(ubPath), R20)
+	// @ fold acc(p.scionLayer.Mem(ub), R7)
+	// @ assert acc(p.path.Mem(ubPath), R20)
+	// TemporaryAssumeForIO(slayers.ValidPktMetaHdr(ub))
+	// TemporaryAssumeForIO(len(absPkt(dp, ub).CurrSeg.Future) > 0)
+	// TemporaryAssumeForIO(p.EqAbsHopField(absPkt(dp, ub)))
```

Comment:
If you haven't started the proof for the fourth assumption, maybe it is a good idea to do that on a separate PR, so that we can keep the PRs small and easy to review